src/frontend/js/hooks/useJoanieUserProfile.tsx

```diff
@@ -0,0 +1,29 @@
+import { defineMessages } from 'react-intl';
+import { API } from 'types/Joanie';
+import { useJoanieApi } from 'contexts/JoanieApiContext';
+import { JoanieUserProfile } from 'types/User';
+import { ResourcesQuery, useResource, UseResourcesProps } from './useResources';
+
+const messages = defineMessages({
+  errorGet: {
+    id: 'hooks.useUser.errorGet',
+    description: 'Error message shown to the user when user fetch request fails.',
+    defaultMessage: 'An error occurred while fetching logged user. Please retry later.',
+  },
+  errorNotFound: {
+    id: 'hooks.useUser.errorNotFound',
+    description: "Error message shown to the user when it isn't logged.",
+    defaultMessage: "You aren't log in.",
+  },
+});
+
+const props: UseResourcesProps<JoanieUserProfile, ResourcesQuery, API['user']['me']> = {
+  queryKey: ['user', 'joanie'],
```

Comment:
As you set session to true, a sessionQuery is used under the hood, so no need to add `user` into the queryKey. Furthermore, from my point of view, `profile` is more explicit than `joanie`
```suggestion
  queryKey: ['profile'],
```